app/routes/AppContainer.jsx

```diff
@@ -0,0 +1,50 @@
+import React, { Component } from "react"
+import PropTypes from "prop-types"
+import { connect } from "react-redux"
+import { ipcRenderer } from "electron"
+
+import {assignmentFetchInfo} from "../modules/assignment/actions/assignment-fetch-info"
+import {setAssignmentURL} from "../modules/assignment/actions/assignment-set-url"
+import {settingsUpdateUserState} from "../modules/settings/actions/settings-update-user-state"
+
+class AppContainer extends Component {
+  constructor (props) {
+    super(props)
+
+    this.props.updateUserState() // Get user if already logged in from previous session
+
+    ipcRenderer.on("open-url", (event, assignmentURL) => {
+      this.props.fetchAssignment(assignmentURL)
+      this.props.router.push({
+        pathname: "/populate",
+      })
+    })
+
+    ipcRenderer.send("initialized")
```

Comment:
If we're interested in signalling when the component is initialized, the constructor feels a bit early in the process to send this. I had a look at [the React component lifecycle](https://reactjs.org/docs/react-component.html#mounting) again and [`componentDidMount()`](https://reactjs.org/docs/react-component.html#componentdidmount) feels like the ideal spot:

> `componentDidMount()` is invoked immediately after a component is mounted (inserted into the tree). 

Yes, this is probably a bit pedantic. Also I might be mis-interpreting what `"initialized"` represents here.

Reply:
Great point, I moved the IPC message sending to the `componentDidMount()` method. The reason I had to do this is because if the app is opened with a deep link, I have to wait for both the Main process to be ready and this Root component to mount before I can send the message to open the Populate page, or else it just gets dropped.